### DISCUSSION TITLE
Easy Question. How to RUN my PdfGenerator Class?

### DISCUSSION THREAD

**Discussion by MoxMorris** [QuestPDF/QuestPDF#897] [Q&A] [answered]
Hi,
i am new in QuestPDF. I was reading the invoice document example [Link](https://www.questpdf.com/getting-started.html). I understood everything, except how to run the damn thing. At the end of the tutorial there is the code for Main method:
![image](https://github.com/QuestPDF/QuestPDF/assets/166315739/265774ab-4a59-488e-a1d6-815778ffa8ba)
The problem is - there is no method **GeneratePdf**. It seems like in the tutorial someone skipped the part where it was implemented...

There is a _Compose_ method, that must recieve an object with type IDocumentContainer and then generate my PDF.
` 
public void Compose(IDocumentContainer container)
 {
     container
         .Page(page =>
        ......
}
`
 In Main method of the tutorial no one uses it, but you must use it to create the PDF, am I right? Then how do you do that? IDocumentContainer is an empty Interface. I tried to create my own class, that implements the empty interface, but surprise, it didn't work. `MyDocument.Compose(new InitializeContainer()); ` and i recieved _System.NullReferenceException: 'Object reference not set to an instance of an object.'_

 
 Everyone in youtube shows the `Document.Create(container => blablabla...)` way, but it is only for very very simple PDFs, and i would want to use the normal classes with my own Models, my own data, etc. In the main video of **JetBrains** from **Marcin Ziabek** there is also the Document.Create way of implementation. 
 
 Please help!

**Comment by MarcinZiabek** [comment, marked as answer]
Thank you for sharing your feedback. I have enhanced the Getting Started tutorial by adding missing `using` statements:

```csharp
using System.IO;
using QuestPDF.Drawing;
using QuestPDF.Fluent;
using QuestPDF.Infrastructure;

static void Main(string[] args)
{
    var model = InvoiceDocumentDataSource.GetInvoiceDetails();
    var document = new InvoiceDocument(model);
    document.GeneratePdfAndShow();
    
    // document.GeneratePdf("invoice.pdf");
}
```

You can also test the entire project by cloning this repository: https://github.com/QuestPDF/QuestPDF

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
